cri/container_create.go

```diff
@@ -45,45 +48,110 @@ func (s *Service) CreateContainer(ctx context.Context, r *criapi.CreateContainer
 
 	config := r.GetConfig()
 	containerName := config.GetMetadata().GetName()
-	log.Debugf("received CreateContainer for %s.", containerName)
 
 	if containerName == userContainerName {
-		var image string
-		// Get image name
-		envs := config.GetEnvs()
-		for _, kv := range envs {
-			if kv.GetKey() == guestImageEnv {
-				image = kv.GetValue()
-				break
-			}
-		}
+		return s.processUserContainer(ctx, r)
+	}
+	if containerName == queueProxyName {
+		return s.processQueueProxy(ctx, r)
+	}
 
-		fi, err := s.coordinator.startVM(context.Background(), image)
-		if err != nil {
-			log.WithError(err).Error("failed to start VM")
-			return nil, err
-		}
+	// Containers irrelevant to user's workload
```

Comment:
please correct this comment. it is very confusing. way what it is relevant to instead of what it is irrelevant to